tools/AzDev/src/Typespec/UpdateTSPModuleCmdlet.cs

```diff
@@ -0,0 +1,784 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using AzDev.Models;
+using NuGet.Protocol.Plugins;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+using System.Net.Http;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace AzDev.Cmdlets.Typespec
+{
+    /*
+        TODO: 
+            0. preinstall tsc/tsp?
+            1. handle additional directories
+            2. add more try catch on file operations
+            3. add more logging
+            4. support debug
+            5. polish the try-finally block
+                5.1 cmdlet could fail before reaching try block which causes temp files not cleaned up
+                5.2 too many lines in the try block make it hard to catch exceptions
+                5.3 maybe use "using + filestream" for IO operations
+    */
+    /*
+        1. -tsplocation provided but no tsp-location.json, update tsp-location with -tsplocation
+        2. -tsplocation provided and tsp-location.yaml exists, use -tsplocation and update tsp-location.yaml
+        3. no -tsplocation and there is a tsp-location.yaml, use tsp-location.yaml as -tsplocation
+        4. if -tspconfig provided, merge it with -tsplocation
+    */
+    [Cmdlet("Update", "DevTSPModule")]
```

Comment:
I prefer to use the verb "set" instead of "update", since this cmdlet is used for both initializing and updating a PS module.